src/components/Icons.ts

```diff
@@ -1,12 +1,14 @@
-import { VueConstructor, CreateElement } from 'vue';
+import { VueConstructor, CreateElement, RenderContext, FunctionalComponentOptions } from 'vue';
+import { DefaultProps } from 'vue/types/options';
 
+// See https://vuejs.org/v2/guide/render-function.html for documentation on createElement and functional components
 // tslint:disable-next-line:variable-name
 const IconBase = (component: VueConstructor) => ({
     functional: true,
-    render(createElement: CreateElement) {
-        return createElement(component, { class: 'nq-icon' });
+    render(createElement: CreateElement, context: RenderContext) {
+        return createElement(component, Object.assign({ class: 'nq-icon' }, context.data));
```

Comment:
Would there be a way to merge the existing `nq-icon` class with other classes set on the element?

Because I think right now, new classes would overwrite the `nq-icon` class, so whenever adding a class to an icon I would have to include the `nq-icon` class right now, right?

Reply:
They get automatically merged by Vue.
Classes specified on the element are specified in `context.staticClass`.